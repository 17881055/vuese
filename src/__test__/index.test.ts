import vuese from '../index'
import * as path from 'path'
import * as fs from 'fs'

function getSource(fileName: string): string {
  const p = path.resolve(__dirname, `../../__fixtures__/${fileName}`)
  const source = fs.readFileSync(p, 'utf-8')
  return source
}

test('The generated document matches the expected results', () => {
  const source = getSource('all.vue')
  expect(vuese(source)).toMatchSnapshot()
})
