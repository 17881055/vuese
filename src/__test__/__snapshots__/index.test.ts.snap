// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The generated document matches the expected results 1`] = `
"|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|name|表单的名称，最多8个字符|\`String\` / \`Number\`|\`true\`|-|

|Event Name|Description|Parameters|
|---|---|---|
|onclear|清空表单时触发| 参数是一个布尔值，代表xxx|

|Name|Description|Default Slot Content|
|---|---|---|
|header|表单头部|\`<th>title</th>\`|

|Method|Description|Parameters|
|---|---|---|
|clear|用来手动清空表单| 参数是一个布尔值，代表xxx|

"
`;
